apache/superset
# fix(Filters): Apply native & cross filters on common columns

### SUMMARY
When in scope (global or specific) charts that are not applicable to the filter (do not share the same datasource or columns) were being applied anyway, causing unnecessary queries. This PR checks that the charts have columns in common or will exclude the charts.

### AFTER

https://github.com/user-attachments/assets/fe58ccbc-b689-4470-a85d-a8b494e9633c


https://github.com/user-attachments/assets/9a45c284-ac44-4f4e-960b-e66c0c5ed46d


### TESTING INSTRUCTIONS
1. Create a native filter that has no common columns with a chart
2. Add the chart in scope
3. Apply the filter
4. The chart should not reload and emit a query

### ADDITIONAL INFORMATION
